initrd/bin/kexec-save-default

```diff
@@ -123,10 +123,8 @@ if [ ! -d $paramsdir ]; then
 fi
 
 if [ "$CONFIG_TPM2_TOOLS" = "y" ]; then
-	if [ ! -r "$PRIMHASH_FILE" ]; then
-		sha256sum /tmp/primary.handle > "$PRIMHASH_FILE" \
-			|| die "ERROR: Failed to Hash TPM2 primary key handle!"
-	fi
+	sha256sum /tmp/primary.handle > "$PRIMHASH_FILE" \
+		|| die "ERROR: Failed to Hash TPM2 primary key handle!"
```

Comment:
This often left a stale version of the primary object handle hash around for me - it'd create the file the first time, but resetting the TPM creates a new primary object, invalidating the hash.  The hash was never updated so it would never match again.

@tlaurion I think this makes sense to do here but I kicked around a few alternatives, interested in your thoughts.  Initially I started deleting kexec_primhdl_hash whenever the TPM was reset (gui-init's TPM reset, OEM reset, reset config), but then thought it does not make sense for kexec-save-default to ever leave an old hash in place anyway.

BTW, OEM reset does later remove kexec*, so that may have worked on an actual board, but qemu doesn't get that far since it can't read back or flash the firmware.